src/platform-aws.cpp

```diff
@@ -1047,3 +1050,38 @@ void PlatformAWS::sort_rails(struct fi_info **info_list, size_t num_rails, size_
 
 	return;
 }
+
+bool PlatformAWS::is_aws() const {
+	if (ofi_nccl_force_non_aws.get()) {
+		NCCL_OFI_INFO(NCCL_INIT, "Disabling PlatformAWS optimizations");
+		return false;
+	}
+
+	hwloc_topology_t topo;
+	if (hwloc_topology_init(&topo) != 0) {
+		return false;
+	}
+
+	// Use shared_ptr for automatic cleanup of hwloc topology on function exit
+	auto topology = std::shared_ptr<hwloc_topology>(topo, hwloc_topology_destroy);
+	/* HWLOC API changes introduced in version 2.0.0 - same pattern as nccl_ofi_topo.cpp */
+#if (HWLOC_API_VERSION >= 0x00020000)
+	hwloc_topology_set_io_types_filter(topo, HWLOC_TYPE_FILTER_KEEP_ALL);
+#else
+	unsigned long flags = hwloc_topology_get_flags(topo);
+	flags |= HWLOC_TOPOLOGY_FLAG_WHOLE_IO;
+	hwloc_topology_set_flags(topo, flags);
+#endif
+	if (hwloc_topology_load(topo) != 0) {
+		return false;
+	}
+
+	hwloc_obj_t obj = nullptr;
+	while ((obj = hwloc_get_next_pcidev(topo, obj)) != nullptr) {
+		if (obj->attr->pcidev.vendor_id == AWS_VENDOR_ID &&
+		    (obj->attr->pcidev.device_id & 0xFFF0) == EFA_DEV) {
```

Comment:
I thought we decided that we needed to look for either an ENA or EFA device?  Otherwise, the topo loading for the G* series won't work.